stable/datadog/values.yaml

```diff
@@ -705,6 +705,12 @@ daemonset:
   #
   #   # Use java cgroup memory awareness
   #   jmx_use_cgroup_memory_limit: true
+  #
+  ## dogstatsd_port - integer - optional - default: 8125
+  ## Override the Agent DogStatsD port.
+  ## Note: Make sure your client is sending to the same UDP port.
+  #
+    dogstatsd_port:
```

Comment:
Your new field should be commented by default.

Reply:
```
  datadog git:(allow-datadog-statsd-port-to-be-variable) ✗ helm template .
Error: render error in "datadog/templates/daemonset.yaml": template: datadog/templates/daemonset.yaml:84:12: executing "datadog/templates/daemonset.yaml" at <include "container-a...>: error calling include: template: datadog/templates/container-agents.yaml:8:42: executing "container-agents" at <.Values.daemonset.cu...>: can't evaluate field dogstatsd_port in type interface {}
➜  datadog git:(allow-datadog-statsd-port-to-be-variable) ✗
```

commenting it out seems to cause me linting issues.
I originally submitted this pr with both `customAgentConfig` and `dogstatsd_port` commented out

Reply:
I think you should not put this new option inside the `customAgentConfig` yaml object.
it will be better to add it after the other `dogstatsd` options here: https://github.com/helm/charts/blob/master/stable/datadog/values.yaml#L149

Reply:
oh interesting, is that ultimately making the same change https://github.com/DataDog/datadog-agent/blob/master/pkg/config/config_template.yaml#L639 would have?

Reply:
@clamoriniere Sorry about that, I linked the wrong line. 
This is the var I'm expecting to change. Am i misinterpreting this?

Reply:
bump :) 
@xlucas  and/or @hkaj 
Would love some additional guidance to get this resolved

Reply:
hi @johnpekcan, `customAgentConfig` is not something we want to itemize. Its value should just be a yaml blob that will then be passed as is in `datadog.yaml`: https://github.com/helm/charts/blob/52e585114f9d4db3f2052321d40f31b8ac3bf969/stable/datadog/templates/datadog-yaml-configmap.yaml#L21-L23

If you want to add a specific option that both configures dogstatsd to listen on that port, and change the port accordingly in the various files you touch, you should make it a new option below `datadog.` as @clamoriniere suggested (one similar option is the one he linked - https://github.com/helm/charts/blob/60683f9b99b721a8e65345383b25045247daea17/stable/datadog/values.yaml#L149 )

Does that make sense?

Reply:
Okay, I think i'm following.
I think I got hung up placing it in that location because locally, that is how I'm configuring the dd binary. 


What is the recommended way to pass this to the binary?
Does there exist an ENV var for DOGSTATSD?
Like https://github.com/helm/charts/blob/60683f9b99b721a8e65345383b25045247daea17/stable/datadog/templates/container-agents.yaml#L23 ?

Reply:
Thanks @hkaj ! Much appreciated!

I think i'm following :) 
I think I got hung up placing it in that location because locally, that is how I'm configuring the dd binary. 

What is the recommended way to pass this to the binary?
Does there exist an ENV var for DOGSTATSD?
Like https://github.com/helm/charts/blob/60683f9b99b721a8e65345383b25045247daea17/stable/datadog/templates/container-agents.yaml#L23 ?

Reply:
yes, you can use the variable: DD_DOGSTATD_PORT

Reply:
excellent, new PR submitted!